webui/react/src/pages/ExperimentDetails/ExperimentCheckpoints.tsx

```diff
@@ -0,0 +1,287 @@
+
+    const { columnKey, order } = tableSorter as SorterResult<CoreApiGenericCheckpoint>;
+    if (!columnKey || !columns.find((column) => column.key === columnKey)) return;
+
+    const newSettings = {
+      sortDesc: order === 'descend',
+      /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+      sortKey: columnKey as any,
+      tableLimit: tablePagination.pageSize,
+      tableOffset: (tablePagination.current - 1) * tablePagination.pageSize,
+    };
+    const shouldPush = settings.tableOffset !== newSettings.tableOffset;
+    updateSettings(newSettings, shouldPush);
+  }, [ columns, settings.tableOffset, updateSettings ]);
+
+  const fetchExperimentCheckpoints = useCallback(async () => {
+    try {
+      const states = (settings.state ?? []).map((state) => (
+        encodeCheckpointState(state as CheckpointState)
+      ));
+      const response = await getExperimentCheckpoints(
+        {
+          id: experiment.id,
+          limit: settings.tableLimit,
+          offset: settings.tableOffset,
+          orderBy: settings.sortDesc ? 'ORDER_BY_DESC' : 'ORDER_BY_ASC',
+          sortBy: validateDetApiEnum(V1GetExperimentCheckpointsRequestSortBy, settings.sortKey),
+          states: validateDetApiEnumList(Determinedcheckpointv1State, states),
+        },
+        { signal: canceler.signal },
+      );
+      setTotal(response.pagination.total ?? 0);
+      setCheckpoints(response.checkpoints);
+    } catch (e) {
+      handleError(e, {
+        publicSubject: `Unable to fetch experiment ${experiment.id} checkpoints.`,
+        silent: true,
+        type: ErrorType.Api,
+      });
+    } finally {
+      setIsLoading(false);
+    }
+  }, [
+    experiment.id,
+    canceler,
+    settings.sortDesc,
+    settings.sortKey,
+    settings.state,
+    settings.tableLimit,
+    settings.tableOffset,
+  ]);
+
+  const submitBatchAction = useCallback(async (action: CheckpointAction) => {
+    try {
+      // TODO: Actions
+
+      // Refetch experiment list to get updates based on batch action.
+      await fetchExperimentCheckpoints();
+    } catch (e) {
+      const publicSubject = `Unable to ${action} Selected Checkpoints`;
+      handleError(e, {
+        level: ErrorLevel.Error,
+        publicMessage: 'Please try again later.',
+        publicSubject,
+        silent: false,
+        type: ErrorType.Server,
+      });
+    }
+  }, [ fetchExperimentCheckpoints ]);
+
+  const { stopPolling } = usePolling(fetchExperimentCheckpoints, { rerunOnNewFn: true });
+
+  // Get new trials based on changes to the pagination, sorter and filters.
+  useEffect(() => {
+    fetchExperimentCheckpoints();
+    setIsLoading(true);
+  }, [
+    fetchExperimentCheckpoints,
+    settings.sortDesc,
+    settings.sortKey,
+    settings.state,
+    settings.tableLimit,
+    settings.tableOffset,
+  ]);
+
+  useEffect(() => {
+    if (terminalRunStates.has(experiment.state)) stopPolling({ terminateGracefully: true });
```

Comment:
i think maybe checkpoints can be deleted (even automatically) after the experiment itself is in a terminal state?